src/arrayMethodJoin.js

```diff
@@ -5,7 +5,31 @@
  */
 function applyCustomJoin() {
   [].__proto__.join2 = function(separator) {
-    // write code here
+    const arrayLength = this.length;
+
+    if (arrayLength === 0) {
+      return '';
+    }
+
+    const finalSeparator = separator === undefined
+      ? ',' : separator === null ? 'null' : separator;
```

Comment:
never nest ternaries. it's unreadable

use function parameters to assign a default value

Reply:
default value assigned to function parameter and nesting removed from ternary.